Recruit-Training/Advanced-Recruit-Training/Viscomm-Teaser/Viscomm-Teaser-Training/src/InformationParser.cpp

```diff
@@ -2,11 +2,113 @@
 
 InformationParser::InformationParser()
 {
-
+    
 }
 
 bool InformationParser::readJSON()
 {
-    //Place Function Code Here
-    return true;
+    QFile inFile("../../CarData.json");
+    inFile.open(QIODevice::ReadOnly|QIODevice::Text);
+    QByteArray data = inFile.readAll();
+    inFile.close();
+
+    QJsonDocument doc = QJsonDocument::fromJson(data);
+    if (doc.isNull()) {
+        return false;
+    }
+
+    QJsonObject json = doc.object();
+    if(json.contains("owner") && json["owner"].isString() &&
+       json.contains("type") && json["type"].isString() &&
+       json.contains("carName") && json["carName"].isString() &&
+       json.contains("batteryPercentage") && json["batteryPercentage"].isDouble() &&
+       json.contains("color") && json["color"].isString() &&
+       json.contains("location") && json["location"].isString() &&
+       json.contains("viscommTeamMembers") && json["viscommTeamMembers"].isArray())
+    {
+        owner = json["owner"].toString();
+        type = json["type"].toString();
+        carName = json["carName"].toString();
+        batteryPercentage = json["batteryPercentage"].toDouble();
+        color = json["color"].toString();
+        location = json["location"].toString();
+
+        QJsonArray teamMemberArray = json["viscommTeamMembers"].toArray();
+        viscommTeamMembers.clear();
+        viscommTeamMembers.reserve(teamMemberArray.size());
+        for(QJsonValueRef a : teamMemberArray)
+        {
+           QJsonObject m = a.toObject();
+           Member member(m["firstName"].toString(), m["lastName"].toString(), m["gradYear"].toString().toInt());
+           viscommTeamMembers.append(member);
+        }
+
+        return true;
+    }
+
+    return false;
+}
+
+QString InformationParser::getOwner() const{
```

Comment:
Remember to follow the style guide at [Style Guide](https://github.com/UCSolarCarTeam/Development-Resources/tree/master/Development-Guide/Style-Guide), which means the curly braces should start on the next line. even for these one-line return functions.